dask/blockwise.py

```diff
@@ -917,21 +919,12 @@ def make_blockwise_graph(
         if deserializing:
             deps.update(func_future_args)
             args += list(func_future_args)
-            if kwargs:
-                val = {
-                    "function": dumps_function(apply),
-                    "args": warn_dumps(args),
-                    "kwargs": warn_dumps(kwargs2),
-                }
-            else:
-                val = {"function": func, "args": warn_dumps(args)}
+        if kwargs:
+            val = (apply, func, args, kwargs2)
```

Comment:
@jrbourbeau and @ian-r-rose - I am able to handle in-lined tasks and avoid the extra `dumps` calls here if I wrap the function in something simple (to make it "callable" again).  However, I am thinking that I am still missing the best way to use the existing serialization machinery (otherwise, the best long-term solution may require improvements in `distributed`/`Serialize`).

The current challenge, both here and in #7417, seems to be that we cannot serialize only the function and/or task arguments (or a subset of them) on the client, and then also serialize the complete task on the scheduler (which is necessary to send the task to a worker).   I would expect that the `serialize` logic would come across a `Serialized` object, and incorperate the header/frames into the *new* `Serialized` object so that calling `deserialize` on the final object would also deserialize the embedded object.  However, for some reason, this does not seem to be the case. (note that I'm still looking into this...)

Reply:
Ah- It looks like this is another case where we just need to add an `iterate_collection=` argument to `serialize`.  If we require the serialization procedure to treat the elements of a collection separately, the nested serialization seems to work for me :)